### PR TITLE
added periodic-file logging

### PR DESCRIPTION
**Pull request by mcjffld**
similar to rotating log file but adds a date stamp to the filename either daily or hourly

### CODE REVIEW

**Comment by trentm** [comment]
I welcome support for this kind of thing. However I was really expecting to have the "rotating with a timestamp" into the "rotating-file" type... with different option to have it use that versus the "count" log suffix.

I'm influenced by my experience with `logadm` (https://smartos.org/man/1m/logadm) on SmartOS. E.g. see the '-t' option in that man page which allows specifying a template with `$foo` params and `%...` codes for strftime params. Esp. this part of that man page:

```
           If no -t option is specified, the default template is $file.$n.
           Actual rotation of log files, where each version is shifted up
           until it expires is done using the $n keyword. If the template does
           not contain the $n keyword, the log file is simply renamed to the
           new name and then the expire rules, if any, are applied.
```

Basically bunyan's `rotating-file` stream right now is implementing that default... with the hope that someday template support would be added, even if only supporting a small subset of fields there... perhaps even diverging to have a quick easy way to get the date-based renaming.

Would you have time to work together on code to get data-based renaming into the existing 'rotating-file'? If so we could design that here or whereever.

**Comment by mcjffld** [comment]
I see what you are talking about but I'm not sold on it.  To me, the basic structure is different between the two.  The rotating log file creates a file and then moves it/them to another name and then creates the original file again.  The proposed periodic file creates unique files every time and there is no file moving required.  I looked at the logadm and I do like that template sort of logic, but it is still for moving files around as opposed to creating date/time stamped log files.

A practical question.  If you are running the app and shut it down for a few hours for server maintenance.  When the app starts back up, what would it do with the existing log file? Would it try to rename the existing log file to the hour that it was when it previous file was created or would it just continue to append to it until the next time to roll logs happened?

**Comment by jsolis** [comment]
I see both sides and I'm a little torn, however, I tend to like the solution proposed by @mcjffld because the logic seems endlessly simpler.  His example is a great one and here is another one: https://github.com/trentm/node-bunyan/issues/117#issuecomment-44804938

Both of those scenarios are cleaner when you just start to write to a new file when your time rolls over.

We will be running node in cluster mode and using Bunyan for logging and I much prefer the new file approach.  I also see the point made in issue 117 about using your system's log rotate instead of Bunyan's built-in file rotation.  IMHO, both are acceptable solutions, but I'm confused by your suggestion:

> Would you have time to work together on code to get data-based renaming into the existing 'rotating-file'? 

because you seem to be suggesting NOT to use the Bunyan's built-in file rotation in your comments in https://github.com/trentm/node-bunyan/issues/117#issuecomment-44804938 at least in the case were cluster mode is being used.

@trentm Would it be reasonable to have this pull request as an option in addition to what you suggested for data-based renaming in the Bunyan log rotate?

**Comment by paambaati** [comment]
+1 :thumbsup:

@trentm Can you please take a look at this?
